script/analysis/restart_flatten_particles.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python
+######################################################################
+# Flattens particle dataset in restart file to new format
+######################################################################
+
+from __future__ import print_function,division
+import h5py
+import numpy as np
+from argparse import ArgumentParser
+
+def get_dsets_and_sizes(h5file):
+    """gets photon datasets and sizes"""
+    keys = []
+    sizes = []
+    for k,v in h5file.items():
+        if 'photons_' in k:
+            keys.append(k)
+            sizes.append(v.shape[0])
+    return keys,sizes
+
+def update_file(h5file):
+    "Updates hdf5 file object"
+    keys,sizes = get_dsets_and_sizes(h5file)
+    if len(keys) <= 0:
+        raise ValueError("This file does not need conversion.")
+    sizes = np.array(sizes)
+    offsets = np.cumsum(sizes)
+    h5file.create_dataset('particle_counts',
+                          data=sizes)
+    h5file.create_dataset('particle_offsets',
+                          data=offsets)
+    data = list([None for k in keys])
+    for i,k in enumerate(keys):
+        print("\tReading dataset",
+              k,"with size",
+              h5file[k].shape)
+        data[i] = h5file[k][:]
+    data = np.hstack(data)
+    h5file.create_dataset('superphotons',
+                          data=data)
```

Comment:
Will this script survive the memory errors that were being seen on some existing restart files? :)

Reply:
We shall see...